### PR TITLE
gnomeExtensions.emoji-selector: init at v19

### PR DESCRIPTION
**Pull request by rawkode**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  uuid = "emoji-selector@maestroschan.fr";
+
+  buildInputs = [ glib ];
+
+  buildPhase = ''
+    glib-compile-schemas ./${uuid}/schemas
+  '';
+
+  installPhase = ''
+    mkdir -p $out/share/gnome-shell/extensions
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix:23]: When overriding `installPhase`, you should also provide `runHook preInstall` and `runHook postInstall`.

File: pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    cp -r ${uuid} $out/share/gnome-shell/extensions
+  '';
+
+  meta = with stdenv.lib; {
+    description =
+      "This GNOME shell extension provides a searchable popup menu displaying most emojis";
+    license = licenses.gpl3;
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix:30]: Looks like `gpl3Plus`: https://github.com/maoschanz/emoji-selector-for-gnome/blob/f67019db7094c1fe2ecf71fcbc458a9ebf52c99b/emoji-selector%40maestroschan.fr/extension.js#L6-L9

File: pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  '';
+
+  installPhase = ''
+    mkdir -p $out/share/gnome-shell/extensions
+    cp -r ${uuid} $out/share/gnome-shell/extensions
+  '';
+
+  meta = with stdenv.lib; {
+    description =
+      "This GNOME shell extension provides a searchable popup menu displaying most emojis";
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix:29]:
```suggestion
      "GNOME Shell extension providing a searchable popup menu displaying most emojis";
```

File: pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{ stdenv, fetchFromGitHub, glib, gettext }:
+
+stdenv.mkDerivation rec {
+  pname = "gnome-shell-emoji-selector";
+  version = "19";
+
+  src = fetchFromGitHub {
+    owner = "maoschanz";
+    repo = "emoji-selector-for-gnome";
+    rev = "${version}";
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix:10]:
```suggestion
    rev = version;
```

File: pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    sha256 = "0x60pg5nl5d73av494dg29hyfml7fbf2d03wm053vx1q8a3pxbyb";
+  };
+
+  uuid = "emoji-selector@maestroschan.fr";
+
+  buildInputs = [ glib ];
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix:16]:
```suggestion
  nativeBuildInputs = [ glib ];
```

**Comment by rawkode** [comment]
Thanks, @jtojnar

File: pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix (hunk 1 above)
Comment (jtojnar) [reply pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix:24]: Same for `buildPhase`.

File: pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix (hunk 2 above)
Comment (jtojnar) [reply pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix:32]: Does not appear resolved.

Comment (rawkode) [reply pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix]: Apologies, I was a bit aggressive with the `git add -p` . Fixed

**Comment by rawkode** [comment]
@jtojnar could I get one last review please? I believe I've addresses these issues

**Comment by jtojnar** [comment]
Looks good. Thanks.

**Pull request merged**